language/golang/language.go

```diff
@@ -106,8 +104,33 @@ func (l *Language) Execute(logger *log.Logger, repositoryPath string) (coverage
 		Directory: repositoryPath,
 	})
 	if err != nil {
-		return 0, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput)
+		if strings.Contains(commandOutput, "ERROR: no test files found") {
+			return 0, problems, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput)
+		}
+
+		testSummary := languageGoTestsErrorMatch.FindStringSubmatch(commandOutput)
+		if len(testSummary) > 0 {
+			testCount, e := strconv.ParseInt(testSummary[1], 10, 64)
+			if e != nil {
+				return 0, problems, pkgerrors.WithStack(e)
+			}
+			errorCount, e := strconv.ParseInt(testSummary[2], 10, 64)
+			if e != nil {
+				return 0, problems, pkgerrors.WithStack(e)
+			}
+			// If there are no executed tests but only errors, then we assume there is a syntax error.
+			if testCount == 0 && errorCount > 0 {
+				return 0, problems, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput)
+			}
+		}
+
+		problems = append(problems, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput))
+	}
+
+	coverage, err = language.CoverageObjectCountOfFile(coverageFilePath)
+	if err != nil {
+		problems = append(problems, pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput))
 	}
```

Comment:
I think if the coverage counting failed, it should still be a hard error, no?

Reply:
Yes. Forgot to change that back in the final version.